cmd/server.go

```diff
@@ -151,7 +158,7 @@ func (s *Server) Run(host, token string, insecure bool) error {
 	s.iam = newIAM(s.BaseRoleARN)
 	r := mux.NewRouter()
 	r.Handle("/{version}/meta-data/iam/security-credentials/", appHandler(s.securityCredentialsHandler))
-	r.Handle("/{version}/meta-data/iam/security-credentials/{role}", appHandler(s.roleHandler))
+	r.Handle("/{version}/meta-data/iam/security-credentials/{role:.*}", appHandler(s.roleHandler))
```

Comment:
Default behaviour is to catch until the next slash (e.g. {role}), this needs to catch an ARN which may or may not have slashes in it hence the regex.